multiqc/modules/nonpareil/nonpareil.py

```diff
@@ -0,0 +1,276 @@
+""" MultiQC module to parse output from nonpareil """
+
+
+import logging
+
+from multiqc import config
+from multiqc.modules.base_module import BaseMultiqcModule, ModuleNoSamplesFound
+
+# Initialise the logger
+from multiqc.plots import linegraph
+
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """Nonpareil analysis is split into two parts: the first (written in C++)
+    performs the subsampling, while the second (written in R) performs the
+    statistical analyses. As such, this model requires the user to post-process
+    the R object from the second part."""
+
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="nonpareil",
+            anchor="nonpareil",
+            href="https://github.com/lmrodriguezr/nonpareil",
+            info="Estimate metagenomic coverage and sequence diversity ",
+            doi="10.1093/bioinformatics/btt584",
+        )
+        # Config options
+        self.plot_observed = getattr(config, "nonpareil", {}).get("plot_observed", True)
+        self.plot_model = getattr(config, "nonpareil", {}).get("plot_model", True)
+        self.disp_type = getattr(config, "nonpareil", {}).get("plot_dispersion", False)
+
+        # Read JSON file
+        self.data_by_sample = dict()
+        for f in self.find_log_files("nonpareil", filehandles=True):
+            json_parsed = self.parse_nonpareil_json(f)
+            common_samples = set(json_parsed).intersection(self.data_by_sample)
+            if common_samples:
+                log.debug(f"Duplicate sample names found ({common_samples})! Overwriting...")
+            self.data_by_sample = self.data_by_sample | json_parsed
```

Comment:
This line breaks for Python 3.8 (see the CI status) as the pipe operator is not supported for dicts in 3.8

```suggestion
            self.data_by_sample.update(json_parsed)
```